multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/auditlogging/impl/AuditLoggingFacadeSLImpl.java

```diff
@@ -22,58 +21,18 @@ public AuditLoggingFacadeSLImpl(DataSource dataSource, UserInfoProvider userInfo
     }
 
     @Override
-    public void logAboutToStart(String action) {
-        String message = MessageFormat.format(Messages.AUDIT_LOG_ABOUT_TO_PERFORM_ACTION, action);
-        writeMessage(auditLogManager.getActionLogger(), message, Level.INFO);
+    public void logDataAccessAuditLog(AuditLogConfiguration configuration) {
+        writeMessage(auditLogManager.getConfigLogger(), configuration.getPerformedAction(), Level.WARN);
     }
 
     @Override
-    public void logAboutToStart(String action, Map<String, Object> parameters) {
-        String message = MessageFormat.format(Messages.AUDIT_LOG_ABOUT_TO_PERFORM_ACTION_WITH_PARAMS, action, parameters);
-        writeMessage(auditLogManager.getActionLogger(), message, Level.INFO);
-    }
-
-    @Override
-    public void logActionStarted(String action, boolean success) {
-        String message = MessageFormat.format(success ? Messages.AUDIT_LOG_ACTION_SUCCESS : Messages.AUDIT_LOG_ACTION_FAILURE, action);
-        writeMessage(auditLogManager.getActionLogger(), message, Level.INFO);
-    }
-
-    @Override
-    public void logConfig(AuditableConfiguration configuration) {
-        String message = MessageFormat.format(Messages.AUDIT_LOG_CONFIG, configuration.getConfigurationType(),
-                                              configuration.getConfigurationName());
-        writeMessage(auditLogManager.getConfigLogger(), message, Level.INFO);
-    }
-
-    @Override
-    public void logConfigUpdate(AuditableConfiguration configuration) {
-        String message = MessageFormat.format(Messages.AUDIT_LOG_UPDATE_CONFIG, configuration.getConfigurationName());
-        writeMessage(auditLogManager.getConfigLogger(), message, Level.INFO);
-    }
-
-    @Override
-    public void logConfigDelete(AuditableConfiguration configuration) {
-        String message = MessageFormat.format(Messages.AUDIT_LOG_DELETE_CONFIG, configuration.getConfigurationName());
-        writeMessage(auditLogManager.getConfigLogger(), message, Level.INFO);
-    }
-
-    @Override
-    public void logConfigCreate(AuditableConfiguration configuration) {
-        String message = MessageFormat.format(Messages.AUDIT_LOG_CREATE_CONFIG, configuration.getConfigurationType(),
-                                              configuration.getConfigurationName());
-        writeMessage(auditLogManager.getConfigLogger(), message, Level.INFO);
-    }
-
-    @Override
-    public void logConfigUpdated(boolean success) {
-        String message = success ? Messages.AUDIT_LOG_CONFIG_UPDATED : Messages.AUDIT_LOG_CONFIG_UPDATE_FAILED;
-        writeMessage(auditLogManager.getConfigLogger(), message, Level.INFO);
+    public void logConfigurationChangeAuditLog(AuditLogConfiguration configuration, ConfigurationChangeActions configurationAction) {
+        writeMessage(auditLogManager.getConfigLogger(), configuration.getPerformedAction(), Level.WARN);
     }
 
     private void writeMessage(Logger logger, String message, Level level) {
```

Comment:
Is this synhcronization relevant? As we have a lot of audit logging, this might be slow

Reply:
while testing i noticed that the audit logging is done in a separte thread. i think this should not be a problem